src/Common/src/System/IO/PathInternal.Windows.cs

```diff
@@ -43,7 +129,7 @@ internal static bool HasIllegalCharacters(string path, bool checkAdditional = fa
             Contract.Requires(path != null);
 
             // Question mark is a normal part of extended path syntax (\\?\)
-            int startIndex = path.StartsWith(LongPathPrefix, StringComparison.Ordinal) ? LongPathPrefix.Length : 0;
+            int startIndex = PathInternal.IsExtended(path) ? ExtendedPathPrefix.Length : 0;
```

Comment:
I did consider it- I was assuming the null check would be trivial compared to the method call and StartsWith cost. Also concerned about this being shared code and easily leaking in a rarely hit null. I'll be paying close attention when I do a perf pass through the scenarios.